### PR TITLE
Handle reverse proxy setting a port in Forwarded-For

### PR DESCRIPTION
**Pull request by Mika56**
As per RFC 7239, a reverse proxy may add a port to the client's IP address:
> If the server receiving proxied requests requires some
> address-based functionality, this parameter MAY instead contain an IP
> address (and, potentially, a port number)
https://datatracker.ietf.org/doc/html/rfc7239.html#section-5.2:~:text=and%2C%20potentially%2C%20a%20port%20number

This MR removes that port from the IP address, allowing the `Forwarde-For` header to be properly handled.

### CODE REVIEW

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -612,6 +612,11 @@ public function getRemoteAddress(): string {
 							$IP = substr($IP, 1, -1);
 						}
 
+						// remove client port when set
+						if(str_contains($IP, ':')) {
```
Comment (nickvergessen) [inline lib/private/AppFramework/Http/Request.php:616]: this completely breaks IPv6

Comment (Mika56) [reply lib/private/AppFramework/Http/Request.php]: Nice catch. I've updated the PR with new code that should work better

**Comment by kesselb** [comment]
As reference, how Symfony solved the same problem: https://github.com/symfony/symfony/blob/d9d6024a1a663caa7f13360a4a1d9de443e9c954/src/Symfony/Component/HttpFoundation/Request.php#L1986-L1997

I recommend adding some automated tests to make sure the parsing does not break in the future.

**Comment by Mika56** [comment]
And the corresponding tests: https://github.com/symfony/symfony/blob/d9d6024a1a663caa7f13360a4a1d9de443e9c954/src/Symfony/Component/HttpFoundation/Tests/RequestTest.php#L1034
Should we use their test suite? I don't see support for `[IPv6]:port`
Would that be adding a file to `tests/lib/AppFramework/Http` ?

**Comment by nickvergessen** [comment]
You can add your tests in https://github.com/nextcloud/server/blob/216b95f8b1c4d3e5bf48e3b6075b727a71d2e30d/tests/lib/AppFramework/Http/RequestTest.php#L552

**Comment by nickvergessen** [comment]
Do you need more help? Or shall we take over to finish it?

**Comment by Mika56** [comment]
I lack time, if you're willing to take it over, please go for it! :)

**Comment by nickvergessen** [comment]
Due to changes in CI a rebase was required, but I was unable to push that. So I made a new PR at https://github.com/nextcloud/server/pull/43552

**Comment by Mika56** [comment]
No problem, thanks for taking care of it!

**Pull request closed without merging**